.github/workflows/ci.yaml

```diff
@@ -37,18 +37,15 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
-
       - name: Install uv
         uses: astral-sh/setup-uv@v5
+        with:
+          python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
         run: |
-          uv pip install --system --upgrade '.[test,docs]'
-          uv pip list --system
+          uv pip install --upgrade '.[test,docs]'
+          uv pip list
```

Comment:
Ah that's interesting. I didn't realize that if you setup Python with `uv` that it knows how to automatically activate the default virtual environment it creates. 

https://github.com/astral-sh/setup-uv/blob/22695119d769bdb6f7032ad67b9bca0ef8c4a174/README.md#python-version

That's good to know, so thanks @ponyisi!